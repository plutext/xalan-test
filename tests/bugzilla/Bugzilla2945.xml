<myDoc>
   <myNode id="tools">
        <myItem>Item of tools</myItem>
   </myNode>
   <myNode id="main">
        <myItem>Item of main</myItem>
   </myNode>
   <myNode id="aux">
        <myItem>Item of aux</myItem>
   </myNode>
</myDoc>
