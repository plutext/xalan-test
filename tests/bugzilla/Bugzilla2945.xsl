<?xml version="1.0" encoding="UTF-8"?>
<xsl:stylesheet version="1.0" 
     xmlns:xsl="http://www.w3.org/1999/XSL/Transform" 
     xmlns:fo="http://www.w3.org/1999/XSL/Format">


<xsl:template match="/">
  <xsl:apply-templates/>
</xsl:template>

<xsl:template match="myNode">
  <xsl:apply-templates select="myItem"/>
</xsl:template>

<xsl:template match="myItem">
  <xsl:variable name="nodeSet" select="ancestor::myNode[@id='main'] |ancestor::myNode[@id='aux']"/>
  <xsl:if test="$nodeSet">
    <xsl:value-of select="normalize-space($nodeSet)"/>
	<xsl:for-each select="preceding::* | dummy">
		<xsl:text>, </xsl:text><xsl:value-of select="name(preceding-sibling::* | dummy)"/>
	</xsl:for-each>
  </xsl:if>
</xsl:template>

</xsl:stylesheet>
