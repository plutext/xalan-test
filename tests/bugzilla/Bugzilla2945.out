<?xml version="1.0" encoding="UTF-8"?>

   
   Item of main, , 
   Item of aux, , , myNode, 
